### ISSUE TITLE
filled() !='' narrowing regresses class-string and non-string inputs

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#771]
## Summary

The `@psalm-assert-if-true !='' \$value` line added to `filled()` in #762 (commit 53e67e3) causes regressions on `class-string<X>` and non-string input types. Observed on `filamentphp/filament` v4.8.3 → v4.8.4 benchmark: **+35 new issues** directly attributable to this assertion (14 `InvalidStringClass`, 7 `TypeDoesNotContainType`, 21 `RedundantCondition` on `filled()`, 6 `UndefinedMethod`/`MixedArgument` on `\$enum::cases()`).

## Reproducer

Idiomatic Laravel code that regresses:

```php
/** @return ?class-string<Cluster> */
public static function getCluster(): ?string { /* ... */ }

if (filled(\$cluster = static::getCluster())) {
    \$cluster::shouldRegisterSubNavigation(); // InvalidStringClass
}
```

Before #762: inside the `if`, `\$cluster` was `class-string<Cluster>`, so `\$cluster::method()` was valid.
After #762: `\$cluster` is narrowed to `non-empty-string`, losing the `class-string<Cluster>` template. Psalm reports `InvalidStringClass`.

Same pattern breaks enum detection via `\$enum::cases()` when `getEnum(): ?class-string<UnitEnum>` is passed through `filled()` then `is_a(..., SomeInterface::class, allow_string: true)`. The `UnitEnum` template is dropped during the `filled()` flattening, so `cases()` resolves only against the narrower interface and is reported as undefined.

## Root cause

`@psalm-assert-if-true !='' \$value` is applied by Psalm as a type assertion that replaces `class-string<X>` with `non-empty-string`, even though `class-string<X>` is already a subtype of `non-empty-string` and the assertion gives no new information.

Additionally, the assertion is being applied to non-string atomics (arrays, enums, objects), producing `TypeDoesNotContainType` / `RedundantCondition` diagnostics on call sites like `filled(\$array)` or `filled(\$enum)` that were previously clean.

## Why the stub is fragile even if Psalm changes

Laravel treats `false` as filled (`blank(false) === false`). The `!=''` assertion on `\$value === false` is violated under PHP loose equality (`false == ''` is `true`). The stub currently stays sound only because Psalm applies the `!=''` assertion narrowly to the string atomic — if Psalm broadens that application in a future release, the stub will start incorrectly removing `false` from `mixed|false` inputs. The commit message for #762 acknowledges this.

## Options

1. **Revert the `!=''` assertion** (recommended). The `?string → string` narrowing from the `!null` assertion alone is still a win, and sidesteps the regressions above. The tighter `?string → non-empty-string` narrowing can be reintroduced later if Psalm grows input-type-guarded assertions.
2. Keep the assertion but restrict it to `?string` via some stub-overload trick. Unclear this is expressible in Psalm's docblock grammar today.
3. File upstream bugs for the `class-string<X> → non-empty-string` flattening and the non-string atomic behavior, then wait. The stub regression would remain in the meantime.

## Benchmark evidence (filament)

Net +63 issues v4.8.3 → v4.8.4, ~35 of which stem from this change:

| Type | Count | Pattern |
|---|---:|---|
| RedundantCondition | 21 | `filled(\$array)` / `filled(\$enum)` |
| InvalidStringClass | 14 | `filled(\$classString = ...) && \$classString::method()` |
| TypeDoesNotContainType | 7 | `filled()` on enum/`Model`/`Closure` |
| UndefinedMethod + MixedArgument | 6 | `\$enum::cases()` after intersection lost |

Related: #762, #753, #755.
